### DISCUSSION TITLE
Column support of mapped_column kwargs

### DISCUSSION THREAD

**Discussion by zent1n0** [sqlalchemy/sqlalchemy#11833] [Ideas]
I'm currently coding orm using sqlmodel bindings. In my case, I need to declare a column like this:
```python
  setting_base: List[Dict[str, Any]] = Field(sa_column=Column(JSON)
```
I want the column to be load after sorting using order_by(), from the documents I found Column Loading Options defer() useful, so I'm setting `deferred = True` arguments in the Column, as:
```python
  setting_base: List[Dict[str, Any]] = Field(sa_column=Column(JSON, defered=True)
```
but got this:
```
TypeError: Additional arguments should be named <dialectname>_<argument>, got 'deferred'
```

while debugging, I found the argument seems to be an `mapped_column` only feature.
```python
# statements that raise an TypeError from SQLAlchemy.aql.base::DialectKWArgs._validate_dialect_kwargs, L595
Column(JSON, deferred=True)
# statements functions normal. originally from sa 2.0 docs
mapped_column(JSON, deferred=True)
```

So I'm wondering is there any plan to support more kwargs for Column back from mapped_column.

**Comment by CaselIT** [comment]
Hi,

I'm -1 on adding orm only features to Column, since orm now has `mapped_column`.
You could try using `deferred(Column(JSON))`.  -- edit meant deferred(), not defer()

If that doesn't work you, then you should ask sqlmodel how to do that

**Reply by zent1n0** [reply]
Tks for your quick reply. `defer(Column(JSON))` doesn't work in ```setting_base: List[Dict[str, Any]] = Field(sa_column=defer(Column(JSON))```, and I have turned to control the column over orm `options(defer())`.

**Reply by CaselIT** [reply]
don't know then. should probably ask sqlmodel how to do that

**Reply by CaselIT** [reply]
I've updated the suggestion above `deferred` not `defer` should be used there

**Comment by zzzeek** [comment]
1. if SQLModel is not using declarative and passing `Column` to `Table`, then there is no parameter that can be added in any case, this is a `Column`.
2. if SQLModel **is** using declarative and passing`Column` to a mapped class as an attribute, then you can pass `deferred()` just as easily.
3. this is 100% a SQLModel issue and you should be asking them

**Reply by zent1n0** [reply]
Tks for your suggestions 😊
I'll figure out which works on my case
